spring-integration-mqtt/src/main/java/org/springframework/integration/mqtt/inbound/MqttPahoMessageDrivenChannelAdapter.java

```diff
@@ -478,4 +503,29 @@ public void acknowledge() {
 
 	}
 
+	private class MessageListener implements IMqttMessageListener, MqttCallback {
+
+		private final IMqttAsyncClient client;
+
+		MessageListener(IMqttAsyncClient client) {
+			this.client = client;
```

Comment:
Doesn't look like this one is used somehow...
On the other hand I wonder what is the reason behind extending an `MqttCallback` for this listener as well?..

Reply:
No reason. Just forgot to revert this change after an experiment. Thanks for pointing, those both things can be removed 🙂